Mathlib/Algebra/Divisibility/Prod.lean

```diff
@@ -0,0 +1,25 @@
+/-
+Copyright (c) 2023 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import Mathlib.Algebra.Divisibility.Basic
+import Mathlib.Algebra.Group.Prod
+
+/-!
+# Lemmas about the divisibility relation in product (semi)groups
+-/
+
+variable {G₁ : Type _} {G₂ : Type _} [Semigroup G₁] [Semigroup G₂]
+
+theorem prod_dvd_iff {x y : G₁ × G₂} :
+    x ∣ y ↔ x.1 ∣ y.1 ∧ x.2 ∣ y.2 := by
+  cases x; cases y
+  simp only [dvd_def, Prod.exists, Prod.mk_mul_mk, Prod.mk.injEq,
+    exists_and_left, exists_and_right, and_self, true_and]
+
+@[simp]
+theorem pair_dvd_pair_iff {x₁ y₁ : G₁} {x₂ y₂ : G₂} :
```

Comment:
Should it be `Prod.mk_dvd_mk`? We have `Prod.mk_mul_mk`, not `pair_mul_pair`. Otherwise LGTM. Thanks!
bors d+